misc/testerina/modules/testerina-runtime/src/main/java/org/ballerinalang/test/runtime/entity/CoverageReport.java

```diff
@@ -188,24 +187,41 @@ private void createReport(final IBundleCoverage bundleCoverage, ModuleCoverage m
                             }
                         }
                         moduleCoverage.addSourceFileCoverage(document, coveredLines, missedLines);
-
                     }
                 }
             }
-
         }
     }
 
     private List<Path> filterPaths(Collection<Path> pathCollection) {
         List<Path> filteredPathList = new ArrayList<>();
-
         for (Path path : pathCollection) {
             if (path.toString().contains(this.module.project().sourceRoot().toString()) &&
                     path.toString().contains(target.cachesPath().toString())) {
                 filteredPathList.add(path);
             }
         }
-
         return filteredPathList;
     }
+
+    public Collection<ISourceFileCoverage> modifySourceFiles(Collection<ISourceFileCoverage> sourcefiles) {
+        Collection<ISourceFileCoverage> modifiedSourceFiles = new ArrayList<>();
+        for (ISourceFileCoverage sourcefile : sourcefiles) {
+            if (sourcefile.getName().contains(BLANG_SRC_FILE_SUFFIX)) {
```

Comment:
Any special reason to check `contains()` over `endsWith()` ? Are we getting inner class file patterns as well?

Reply:
No. Can do the check with ```endsWith```. I'll make the changes

Reply:
Fixed in https://github.com/ballerina-platform/ballerina-lang/pull/28767/commits/b33977232ac9c95a1b70e8307c3c9eb5133bde89